src/m3ninx/search/searcher/field.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package searcher
+
+import (
+	"github.com/m3db/m3/src/m3ninx/index"
+	"github.com/m3db/m3/src/m3ninx/postings"
+	"github.com/m3db/m3/src/m3ninx/search"
+)
+
+type fieldSearcher struct {
+	field    []byte
+	compiled index.CompiledRegex
+}
+
+// NewFieldSearcher returns a new searcher for finding documents which match the given field.
+func NewFieldSearcher(field []byte) (search.Searcher, error) {
+	compiledAll, err := index.CompileRegex([]byte(".*"))
+	if err != nil {
+		return nil, err
+	}
+
+	return &fieldSearcher{
+		field:    field,
+		compiled: compiledAll,
+	}, nil
+}
+
+func (s *fieldSearcher) Search(r index.Reader) (postings.List, error) {
+	// TODO: expose a new method on the reader to support such operations
+	return r.MatchRegexp(s.field, s.compiled)
```

Comment:
Ah I see the todo now :p

Reply:
lol, yep - fixed in https://github.com/m3db/m3/pull/1518. this is just to avoid landing a query that doesn't work.